Technolab.OnlineLibrary.Web/Controllers/AuthController.cs

```diff
@@ -60,13 +62,50 @@ private async Task PerformLogin(User user)
                 new Claim(type: Consts.Claim.FirstName, value: user.FirstName),
                 new Claim(type: Consts.Claim.LastName, value: user.LastName),
             };
-            var identity = new ClaimsIdentity(claims, CookieAuthenticationDefaults.AuthenticationScheme);            
-            
+            var identity = new ClaimsIdentity(claims, CookieAuthenticationDefaults.AuthenticationScheme);
+
             await HttpContext.SignInAsync(
                 CookieAuthenticationDefaults.AuthenticationScheme,
                 new ClaimsPrincipal(identity));
         }
 
         private ILibraryDbContextFactory ContextFactory { get; }
+
+        private bool VerifyPassword(string enteredPassword, string correctHashWithParams)

```

Comment:
It's probably worth moving both this and the `GenerateHash` methods to the `User` class itself